### PR TITLE
Fixes for delay()  (issue #36)

### PR DESCRIPTION
**Pull request by iraikov**
This pull request fixes delay() and adds a delay differential equation example to the basics library. However, in order to get the dassl backend to work with this example, the flag info[16] needs to be set to zero. I have not included this last change in the pull request.

### CODE REVIEW

File: src/simcreation.jl [hunk 1]
```diff
@@ -245,7 +245,7 @@ function setup_functions(sm::Sim)
             # to eval globally for two reasons: (1) performance and (2) so
```
Comment (tshort) [inline src/simcreation.jl]: We should zap out-of-date comments like that.

**Comment by tshort** [comment]
Thanks, @iraikov. It looks like you didn't have to go too crazy to add this functionality. Right now, the tests don't run. With info[16] at 1, it fails on your new DDE example. Changing that to 0 causes one of the Tiller examples to fail. Would you mind taking a crack at converting info[16] to an optional argument for `sim`?

**Comment by iraikov** [comment]
Hi Tom, 
I have added a new flag, `alg` which when set to true includes the algebraic variables in the error estimate, and defaults to false.

**Comment by tshort** [comment]
Thanks, Ivan! After a bit more reading in the Sundials docs, we should
probably default to true. -Tom

**Comment by iraikov** [comment]
Ok, I've changed the default to true, meaning that algebraic variables are included in the error estimate.

**Pull request merged**